Add status page and change template from green to black

BankaiNoJutsu opened the pull request:
to:
cc: @subspacecommunity/subspace-maintainers
related to: https://github.com/subspacecommunity/subspace/issues/68
resolves: https://github.com/subspacecommunity/subspace/issues/68

## Background

Adding the status page, and new template

### Changes

- Added status page
- Changed template from green to black

## Testing

Testing has been done locally and with docker on debian, and docker (alpine?) by @jack1902.
Applied mentioned changes in https://github.com/subspacecommunity/subspace/issues/68 by @jack1902.

Not ready for merge yet. See https://github.com/subspacecommunity/subspace/pull/83#issuecomment-639686664

vojta7 commented on handlers.go:
```diff
@@ -281,55 +567,262 @@ func deleteProfileHandler(w *Web) {
+}
+
+func statusHandler(w *Web) {
+	wg_dump, err := exec.Command("wg", "show", "all", "dump").Output()
```
All other handlers use wg0, displayed clients should be only from wg0 to.
```go
	wg_dump, err := exec.Command("wg", "show", "wg0", "dump").Output()
```

vojta7 replied:
I did not pay attention to the rest of the code and missed that you are using the interface name later. But is it really useful when we can't specify an interface name when creating clients? Maybe it would be better to allow global per subspace configuration of interface name to enable running multiple containers at the same time.

vojta7 commented:
I got it working. I had to create config directory in my dir mounted to docker as data in addition to changes mentioned in #68.

Thank you for your work, I really like it :+1:

jack1902 commented:
So I really like the status page additions and the change of the theme but I am going to hold off on approving this for now due to wanting the other pr to be merged.

Once that is merged, I'll take your work and add it to the latest master as I feel there are a large number of commits here to try and address issues that shouldn't exist (issues with numerous forks becoming one)

Also, i would like to have the ability to choose which colour to choose (painful I know, but worth it)

I'll be silent for a few days as it's the weekend but next week I'll assist

BankaiNoJutsu commented:
> So I really like the status page additions and the change of the theme but I am going to hold off on approving this for now due to wanting the other pr to be merged.
> 
> Once that is merged, I'll take your work and add it to the latest master as I feel there are a large number of commits here to try and address issues that shouldn't exist (issues with numerous forks becoming one)
> 
> Also, i would like to have the ability to choose which colour to choose (painful I know, but worth it)
> 
> I'll be silent for a few days as it's the weekend but next week I'll assist

No worries! I just tested locally and in docker. I'm almost done. Just need to fix the routing "allowed ips" or "any" and i'm good waiting for your feedback, and merging with the current master after those waiting pr's.

I agree on the theme selection, however I have no clue how we can implement this. I guess with substitution of the color with a scss variable of some sort.

I may "rebase" my fork, to make the commits cleaner, as it's messy how i did it. sorry.

Have a great weekend!

BankaiNoJutsu commented:
> I got it working. I had to create config directory in my dir mounted to docker as data in addition to changes mentioned in #68.
> 
> Thank you for your work, I really like it +1

Glad you like it!

BankaiNoJutsu commented:
Hey, it's rainy here, so i fixed a few things, and added a theme selection.
So far I have just added mine and the default one. Updated the README as well.

Cheers!

BankaiNoJutsu commented:
Sorry, stupid question, but how can i change my pull request to only include commit https://github.com/subspacecommunity/subspace/pull/83/commits/f76c9353aea4a9e51496f34de279bb3897bf1090 and newer?

jack1902 commented:
@BankaiNoJutsu i'm going to create a new branch moving the relevant parts of this PR over as #75 has now been merged. Will update once its done :)

BankaiNoJutsu commented:
@jack1902 yep! I think that everything should work. But i could also rebase, and create a new pull, it would be cleaner and only necessary commits?
As you wish.

jack1902 commented:
So i am currently looking into a better method using CSS to change the colour, as the seperate folder is not maintainable going forward.

I am not a frontend dev and am very busy with work currently so struggeling to find time. Once i have some though, i will do what i can
